### PR TITLE
Error page dash fix

### PR DESCRIPTION
**Pull request by stefanhar**
# \<Description\>

https://app.asana.com/0/1199123945323117/1199693578639224

## What

- Made route template regex match dash
- Added additional test for this case
- Stopped using type resolver on error page

## Why

- Dash in url causes 500 error on error page
- Make the type resolver more robust

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/hooks/useLinkResolver/useLinkResolver.spec.tsx [hunk 1]
```diff
@@ -163,14 +163,28 @@ describe('Type resolver', () => {
...
   })
+
+  it('Should resolve paths with dashes', () => {
+    const types = typeResolver('/andes-foo/andes-bar')
+    expect(types).toEqual({
+      type: 'subarticle',
+      locale: 'is',
+      slug: ['andes-foo', 'andes-bar'],
+    })
+  })
```
Comment (siggakogk) [inline apps/web/hooks/useLinkResolver/useLinkResolver.spec.tsx:187]: nice!

**Review by siggakogk — APPROVED** [review]
LGTM

**Comment by AndriMar** [comment]
If I understand this correctly this is only fixing the `-` character and not all other special and icelandic characters `.` `~` `æ`
I think this fix needs to be more generic if I understand this correctly.

**Comment by eirikurn** [comment]
These were separate issues. The other one is an internal nextjs bug which is tricky to work around. I think it's been fixed in the latest nextjs and we'll upgrade during next refactor week.

**Comment by AndriMar** [comment]
> These were separate issues. The other one is an internal nextjs bug which is tricky to work around. I think it's been fixed in the latest nextjs and we'll upgrade during next refactor week.

Ok let's see how this will behave after the nextjs update.

**Pull request merged**